packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -314,6 +314,16 @@ export class ApolloServerBase {
       // Don't add the extension here (we want to add it later in generateSchemaDerivedData).
     }
 
+    if (gateway && typeof engine === 'object' && engine.experimental_schemaReporting) {
```

Comment:
This doesn't include the case where someone uses the environment variable. I think it would be less brittle if we make `schemaReporting` boolean public on the agent, and then check if that is true.

Reply:
Good catch!